SubZero/src/model/FPGAModel.h

```diff
@@ -9,12 +9,15 @@
 #define FPGAMODEL_H_
 
 #include "Model.h"
+#include "FPGAState.h"
 
 /**
  * This is a concrete child of the Model class. It implements functions
  * specific to communicating with the FGPA.
  */
 class FPGAModel: public Model {
+private:
+	FPGAState* fpgaState;
```

Comment:
Instead of each concrete Model class having its own different reference to a State, have a generic State class inside the generic Model class.  Then in your concrete Model classes, you can dynamically type cast your generic State reference to the specific concrete State class.